File: arrays_hashing/contains_duplicate.py
# own solution
# Time: O(n)
# Memory: O(n)
class Solution:
    def containsDuplicate(self, nums: list[int]) -> bool:
        return not len(nums) == len(set(nums))

# More efficient solution
# doesn't have to iterate through and store every unique element in nums
# Time: O(n)
# Memory: O(n)
class Solution:
    def containsDuplicate(self, nums: list[int]) -> bool:
        hashset = set()
        
        for n in nums:
            if n in hashset:
                return True
            hashset.add(n)
            
        return False
            

File: arrays_hashing/contains_duplicate.cpp
#include <vector>
#include <unordered_set>

using namespace std;

// Time: O(n)
// Space: O(n)

class Solution {
public:
    bool containsDuplicate(vector<int>& nums) {
        unordered_set<int> hashset;
        for (int E : nums) {
            if (hashset.count(E) > 0) {
                return true;
            }
            hashset.insert(E);
        }
        return false;
    }
};
